File: spec/cpp_stl/test_str_encodings.cpp
#include <boost/test/unit_test.hpp>

#include <str_encodings.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_str_encodings) {
    std::ifstream ifs("src/str_encodings.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    str_encodings_t* r = new str_encodings_t(&ks);

    BOOST_CHECK_EQUAL(r->str1(), "Some ASCII");
    BOOST_CHECK_EQUAL(r->str2(), "こんにちは");
    BOOST_CHECK_EQUAL(r->str3(), "こんにちは");
    BOOST_CHECK_EQUAL(r->str4(), "░▒▓");

    delete r;
}

File: spec/rust/tests/test_str_encodings.rs
#![allow(dead_code)]
use std::fs;

extern crate kaitai;
use self::kaitai::*;
mod formats;
use formats::str_encodings::*;

#[test]
fn test_str_encodings() {
    let bytes = fs::read("../../src/str_encodings.bin").unwrap();
    let reader = BytesReader::new(&bytes);
    let mut r = StrEncodings::default();

    if let Err(err) = r.read(&reader, None, Some(KStructUnit::parent_stack())) {

        panic!("{:?}", err);
    }
    assert_eq!(r.str1(), "Some ASCII");
    assert_eq!(r.str2(), "\u{3053}\u{3093}\u{306b}\u{3061}\u{306f}");
    assert_eq!(r.str3(), "\u{3053}\u{3093}\u{306b}\u{3061}\u{306f}");
    assert_eq!(r.str4(), "\u{2591}\u{2592}\u{2593}");
}
